docs/book/src/clusteropenstack/configuration.md

```diff
@@ -387,7 +387,70 @@ spec:
 
 Any such ports are created in addition to ports used for connections to networks or subnets.
 
-Also, `port security` can be applied to specific port to enable/disable the `port security` on that port; When not set, it takes the value of the corresponding field at the network level.
+### Port network and IP addresses
+
+Together, `network` and `fixedIPs` define the network a port will be created on, and the addresses which will be assigned to the port on that network.
+
+`network` is a filter which uniquely describes the Neutron network the port will be created be on. Machine creation will fail if the result is empty or not unique. If a network `id` is specified in the filter then no separate OpenStack query is required. This has the advantages of being both faster and unambiguous in all circumstances, so it is the preferred way to specify a network where possible.
+
+The available fields are described in [the CRD](https://doc.crds.dev/github.com/kubernetes-sigs/cluster-api-provider-openstack/infrastructure.cluster.x-k8s.io/OpenStackMachine/v1alpha6@v0.7.1#spec-ports-network).
+
+If `network` is not specified at all, it may be possible to infer the network from any uniquely defined subnets in `fixedIPs`. As this may result in additional OpenStack queries and the potential for ambiguity is greater, this is not recommended.
+
+`fixedIPs` describes a list of addresses from the target `network` which will be allocated to the port. A `fixedIP` is either a specific `ipAddress`, a `subnet` from which an ip address will be allocated, or both. If only `ipAddress` is specified, it must be valid in at least one of the subnets defined in the current network. If both are defined, `ipAddress` must be valid in the specified subnet.
+
+`subnet` is a filter which uniquely describe the Neutron subnet an address will be allocated from. Its operation is analogous to `network`, described above.
+
+`fixedIPs`, including all fields available in the `subnet` filter, are described in [the CRD](https://doc.crds.dev/github.com/kubernetes-sigs/cluster-api-provider-openstack/infrastructure.cluster.x-k8s.io/OpenStackMachine/v1alpha6@v0.7.1#spec-ports-fixedIPs).
+
+If no `fixedIPs` are specified, the port will get an address from every subnet in the network.
```

Comment:
`every subnet ` => is this first subnet of the network? This seems to be the way I provide my openstack to my end user..

Reply:
I tested this few days ago, if network has multiple subnets, creating a port specifying just the network gives you a port with address in every of the network's subnets:

A POST with:
```
{
  "port": {
    "admin_state_up": true,
    "name": "test",
    "network_id": "6acb08e4-86a1-47fa-a806-2f79a0b48abb"
  }
}
```

Will get you:
```
{
  "port": {
    <SNIP>
    "fixed_ips": [
      {
        "ip_address": "192.168.25.127",
        "subnet_id": "4b73762e-8fbb-4555-9df6-fcd2b8fe2734"
      },
      {
        "ip_address": "2001:db8::2c8",
        "subnet_id": "d808f460-99b0-4356-b158-92c3b6f97efd"
      }
    ],
    "id": "f5d15252-9f61-4d95-885b-9c3346fa2608",
    <SNIP>
    "name": "test",
    "network_id": "6acb08e4-86a1-47fa-a806-2f79a0b48abb",
    <SNIP>
  }
}
```